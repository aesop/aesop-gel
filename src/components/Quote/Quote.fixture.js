export default {
  author: 'Lorem ipsum',
  content:
    '‘Lorem ipsum dolor sit amet, consectetur adipiscing sit amet ipsum dolor elit’',
};
