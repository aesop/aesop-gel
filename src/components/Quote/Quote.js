import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import styles from './Quote.module.css';

const Quote = ({ author, className, content, hasTopPadding }) => {
  const classSet = cx(
    styles.base,
    { [styles.noTopPadding]: !hasTopPadding },
    className,
  );

  return (
    <div className={classSet}>
      <div className={styles.wrapper}>
        <blockquote className={styles.blockquote}>{content}</blockquote>
        <cite className={styles.author}>{author}</cite>
      </div>
    </div>
  );
};

Quote.propTypes = {
  author: PropTypes.string.isRequired,
  className: PropTypes.string,
  content: PropTypes.string.isRequired,
  hasTopPadding: PropTypes.bool,
};

Quote.defaultProps = {
  author: undefined,
  className: undefined,
  content: undefined,
  hasTopPadding: true,
};

export default Quote;
