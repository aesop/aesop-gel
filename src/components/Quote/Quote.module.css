@import 'styles/index';

.base {
  padding: 60px var(--layout-sm-spacing);

  @media (--viewport-md) {
    padding: 100px var(--layout-md-spacing);
  }

  @media (--viewport-lg) {
    padding: 160px var(--layout-lg-spacing);
  }
}

.dark {
  @extend %font-theme-dark-copy;
}

.light {
  @extend %font-theme-light-copy;
}

.wrapper {
  margin: 0 auto;
  text-align: center;

  @media (--viewport-md) {
    width: 50%;
    margin: 0 auto;
  }
}

.blockquote {
  padding: 0;
  margin: 0;
  margin-bottom: 20px;
  font-family: var(--font-family-zapf-serif);
  font-size: rem(25px);
  font-weight: var(--font-weight-normal);
  line-height: 1.33;
  text-size-adjust: 100%;

  @media (--viewport-md) {
    font-size: rem(30px);
  }
}

.author {
  display: block;
  margin: 0;
  font-family: var(--font-family-sans-serif-medium);
  font-size: rem(14px);
  font-style: initial;
  line-height: 2;
}
