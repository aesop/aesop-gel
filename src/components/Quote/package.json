{
  "name": "Quote",
  "version": "0.0.0",
  "main": "Quote.js"
}
