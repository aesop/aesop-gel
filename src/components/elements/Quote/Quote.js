import React from 'react';
import cx from 'classnames';
import Props from './Quote.types';
import styles from './Quote.module.css'; // @TODO Address this

const Quote: React.FunctionComponent<Props> = ({
  author,
  children,
  classSet,
  noTopPadding,
}) => {
  const className: string = cx(
    styles.base,
    { [styles.noTopPadding]: noTopPadding },
    classSet,
  );

  return (
    <div className={className}>
      <div className={styles.wrapper}>
        <blockquote className={styles.blockquote}>
          <p>{children}</p>
        </blockquote>
        <cite className={styles.author}>{author}</cite>
      </div>
    </div>
  );
};

Quote.defaultProps = {
  noTopPadding: false,
};

export default Quote;
