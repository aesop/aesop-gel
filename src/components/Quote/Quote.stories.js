import React from 'react';
import { storiesOf } from '@storybook/react';
import { text } from '@storybook/addon-knobs';
import '~.storybook/storybook.module.css';
import Quote from './Quote';
import QuoteFixture from './Quote.fixture';

storiesOf('Components/Quote', module).add('Base component', () => (
  <Quote
    author={text('author', QuoteFixture.author)}
    content={text('children', QuoteFixture.content)}
  />
));
