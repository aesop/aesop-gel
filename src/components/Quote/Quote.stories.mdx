import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import * as knobs from '@storybook/addon-knobs';
import Quote from './Quote';
import QuoteFixture from './Quote.fixture';

<Meta title="Components/Quote" component={Quote} />

# Quote

<Preview>
  <Story name="Base component">
    <Quote
      author={knobs.text('author', QuoteFixture.author, 'Copy')}
      content={knobs.text('children', QuoteFixture.content, 'Copy')}
      theme={knobs.select('theme', ['dark', 'light'], 'dark', 'Presentation')}
    />
  </Story>
</Preview>

## API

<Props of={Quote} />
