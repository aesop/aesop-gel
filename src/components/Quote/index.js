/* istanbul ignore file */
import { Quote } from './Quote';

export { Quote };
