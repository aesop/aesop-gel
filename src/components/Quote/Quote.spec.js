import React from 'react';
import renderer from 'react-test-renderer';
import Quote from './Quote';
import QuoteFixture from './Quote.fixture';

describe('<Quote />', () => {
  it('should be defined', () => {
    expect(Quote).toBeDefined();
  });

  it('renders base component correctly', () => {
    const tree = renderer
      .create(
        <Quote author={QuoteFixture.author} content={QuoteFixture.content} />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
