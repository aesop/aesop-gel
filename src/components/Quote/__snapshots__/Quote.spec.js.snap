// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quote /> renders base component correctly 1`] = `
<div
  className="base"
>
  <div
    className="wrapper"
  >
    <blockquote
      className="blockquote"
    >
      ‘Do not wish to be anything but what you are, and try to be that perfectly’
    </blockquote>
    <cite
      className="author"
    >
      Diana Ramirez
    </cite>
  </div>
</div>
`;
